src/App.js

```diff
@@ -63,13 +91,20 @@ class App extends Component {
                   <CloseIcon className={ classes.leftIcon }/>
                   Deselect
                 </Button>
+                <Button variant="contained" color="secondary" className={ classes.leftIcon }
+                        onClick={ this.removeFolder }>
+                  <DeleteIcon className={classes.iconButton} />
+                  Delete
+                </Button>
                 { selectedCommand.label }
               </div>
             ) : (
               <CommandViewEdit
                 command={ selectedCommand }
                 onUpdateCommand={ updateCommand }
-                onRunCommand={runCommand}
+                onRunCommand={ runCommand }
+                onDeleteCommand={ deleteCommand }
+                openConfirmation={ (callback) => {this.openConfirmation(callback)}}
```

Comment:
here you already know that this will delete the `selectedCommand `. sending `deleteCommand` and sending it back as a callback is not necessary. just do something like 'on delete command open the dialog'. the dialog will than execute the delete on confirm

this will remove the need for `this.runCallback` and `this.removeFolder`